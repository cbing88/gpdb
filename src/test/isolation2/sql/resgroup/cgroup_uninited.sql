-- resgroup is enabled but cgroup is not properly setup yet

-- start_ignore
! gpconfig -r gp_resource_manager;
! gpstop -rai;
! rmdir /sys/fs/cgroup/cpu/gpdb/*/;
! rmdir /sys/fs/cgroup/cpuacct/gpdb/*/;
-- to make vim happy */
! rmdir /sys/fs/cgroup/cpu/gpdb;
! rmdir /sys/fs/cgroup/cpuacct/gpdb;
-- end_ignore

-- gpdb cgroup is missing
! gpconfig -c gp_resource_manager -v group;

-- start_ignore
! mkdir /sys/fs/cgroup/cpu/gpdb;
! mkdir /sys/fs/cgroup/cpuacct/gpdb;
! chmod 644 /sys/fs/cgroup/cpu/gpdb;
! chmod 644 /sys/fs/cgroup/cpuacct/gpdb;
-- end_ignore

-- no enough permission on gpdb cgroup
! gpconfig -c gp_resource_manager -v group;

-- start_ignore
! chmod 755 /sys/fs/cgroup/cpu/gpdb;
! chmod 755 /sys/fs/cgroup/cpuacct/gpdb;
! chmod 444 /sys/fs/cgroup/cpu/gpdb/cgroup.procs;
-- end_ignore

-- no enough permission on gpdb cgroup's config files
! gpconfig -c gp_resource_manager -v group;

-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb;
! rmdir /sys/fs/cgroup/cpuacct/gpdb;
! gpconfig -r gp_resource_manager;
-- end_ignore

