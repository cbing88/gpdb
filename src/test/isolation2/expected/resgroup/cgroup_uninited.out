-- resgroup is enabled but cgroup is not properly setup yet

-- start_ignore
! gpconfig -r gp_resource_manager;
20170426:21:13:05:016712 gpconfig:sdw6:gpadmin-[INFO]:-completed successfully

! gpstop -rai;
20170426:21:13:05:016787 gpstop:sdw6:gpadmin-[INFO]:-Starting gpstop with args: -rai
20170426:21:13:05:016787 gpstop:sdw6:gpadmin-[INFO]:-Gathering information and validating the environment...
20170426:21:13:05:016787 gpstop:sdw6:gpadmin-[INFO]:-Obtaining Greenplum Master catalog information
20170426:21:13:05:016787 gpstop:sdw6:gpadmin-[INFO]:-Obtaining Segment details from master...
20170426:21:13:06:016787 gpstop:sdw6:gpadmin-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 5.0.0-alpha.1+dev.100.g5716465 build dev'
20170426:21:13:06:016787 gpstop:sdw6:gpadmin-[INFO]:-There are 0 connections to the database
20170426:21:13:06:016787 gpstop:sdw6:gpadmin-[INFO]:-Commencing Master instance shutdown with mode='immediate'
20170426:21:13:06:016787 gpstop:sdw6:gpadmin-[INFO]:-Master host=sdw6
20170426:21:13:06:016787 gpstop:sdw6:gpadmin-[INFO]:-Commencing Master instance shutdown with mode=immediate
20170426:21:13:06:016787 gpstop:sdw6:gpadmin-[INFO]:-Master segment instance directory=/data1/tpz/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
20170426:21:13:07:016787 gpstop:sdw6:gpadmin-[INFO]:-Attempting forceful termination of any leftover master process
20170426:21:13:07:016787 gpstop:sdw6:gpadmin-[INFO]:-Terminating processes for segment /data1/tpz/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
20170426:21:13:07:016787 gpstop:sdw6:gpadmin-[INFO]:-No standby master host configured
20170426:21:13:07:016787 gpstop:sdw6:gpadmin-[INFO]:-Commencing parallel primary segment instance shutdown, please wait...
20170426:21:13:07:016787 gpstop:sdw6:gpadmin-[INFO]:-0.00% of jobs completed
20170426:21:13:17:016787 gpstop:sdw6:gpadmin-[INFO]:-100.00% of jobs completed
20170426:21:13:17:016787 gpstop:sdw6:gpadmin-[INFO]:-Commencing parallel mirror segment instance shutdown, please wait...
20170426:21:13:17:016787 gpstop:sdw6:gpadmin-[INFO]:-0.00% of jobs completed
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-100.00% of jobs completed
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-----------------------------------------------------
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-   Segments stopped successfully      = 6
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-   Segments with errors during stop   = 0
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-----------------------------------------------------
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-Successfully shutdown 6 of 6 segment instances 
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-Database successfully shutdown with no errors reported
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-Cleaning up leftover gpmmon process
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-No leftover gpmmon process found
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-Cleaning up leftover gpsmon processes
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-No leftover gpsmon processes on some hosts. not attempting forceful termination on these hosts
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-Cleaning up leftover shared memory
20170426:21:13:27:016787 gpstop:sdw6:gpadmin-[INFO]:-Restarting System...

! rmdir /sys/fs/cgroup/cpu/gpdb/*/;

-- */
! rmdir /sys/fs/cgroup/cpu/gpdb;

-- end_ignore

-- gpdb cgroup is missing
! gpconfig -c gp_resource_manager -v group;
20170426:21:13:33:017298 gpconfig:sdw6:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170426:21:13:33:017298 gpconfig:sdw6:gpadmin-[CRITICAL]:-new GUC value failed validation: cgroup is not properly configured on segments: sdw6
new GUC value failed validation: cgroup is not properly configured on segments: sdw6


-- start_ignore
! mkdir /sys/fs/cgroup/cpu/gpdb;

! chmod 644 /sys/fs/cgroup/cpu/gpdb;

-- end_ignore

-- no enough permission on gpdb cgroup
! gpconfig -c gp_resource_manager -v group;
20170426:21:13:33:017351 gpconfig:sdw6:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170426:21:13:33:017351 gpconfig:sdw6:gpadmin-[CRITICAL]:-new GUC value failed validation: cgroup is not properly configured on segments: sdw6
new GUC value failed validation: cgroup is not properly configured on segments: sdw6


-- start_ignore
! chmod 755 /sys/fs/cgroup/cpu/gpdb;

! chmod 444 /sys/fs/cgroup/cpu/gpdb/cgroup.procs;

-- end_ignore

-- no enough permission on gpdb cgroup's config files
! gpconfig -c gp_resource_manager -v group;
20170426:21:13:34:017404 gpconfig:sdw6:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170426:21:13:34:017404 gpconfig:sdw6:gpadmin-[CRITICAL]:-new GUC value failed validation: cgroup is not properly configured on segments: sdw6
new GUC value failed validation: cgroup is not properly configured on segments: sdw6


-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb;

! gpconfig -r gp_resource_manager;
20170426:21:13:34:017456 gpconfig:sdw6:gpadmin-[INFO]:-completed successfully

-- end_ignore

